tools/harness-thci/OpenThread.py

```diff
@@ -1363,22 +1363,10 @@ def powerUp(self):
 
     def reset_and_wait_for_connection(self, timeout=3):
         print("Waiting after reset timeout: {} s".format(timeout))
-        start_time = time.time()
         self.__sendCommand('reset', expectEcho=False)
         self.isPowerDown = True
 
-        while time.time() < start_time + timeout:
-            time.sleep(0.3)
-            if not self.IsBorderRouter:
-                self._disconnect()
-                self._connect()
-            try:
-                self.__executeCommand('state', timeout=0.1)
-                break
-            except Exception:
-                continue
-        else:
-            raise AssertionError("Could not connect with OT device {} after reset.".format(self))
+        self.sleep(timeout)
```

Comment:
any background on this change?


I thought that the reconnection logic was helpful.

Reply:
The Leader 5.5.2 and 5.5.3 were failing consistently due to hitting the break immediately.  The test was never sleeping for the required amount of 140 seconds because it never made it through the while loop more than the 1 time.

Reply:
I'm not really aware of the history of this code or what it was meaning to accomplish.  It may be that it was helpful in some other way but I'm not aware of any issues it has resolved.  I only know that the current state of these 2 tests in openthread (from my perspective) is failing.  This change would fix these 2 failures.

Reply:
I think @edmont added the old logic.

Reply:
BTW, I think the original code tend to leave (break) once the device is responsive again.

Reply:
Okay, the problem here is in the implementation of `resetAndRejoin`. It shouldn't make use of `reset_and_wait_for_connection` but just something like:
```
self.powerDown()
time.sleep(timeout)
self.powerUp()
```
This way `powerUp` can be left untouched.

Reply:
@simonlingoogle Do you have any objections?

Reply:
Agree that `reset_and_wait_for_connection` should exit as long as the connection is ready after reset.